test/test_distributions.py

```diff
@@ -1367,12 +1369,17 @@ def setUp(self):
     def test_kl_monte_carlo(self):
         set_rng_seed(0)  # see Note [Randomized statistical tests]
         for (p, _), (_, q) in self.finite_examples:
-            x = p.sample(sample_shape=(100000,))
-            expected = (p.log_prob(x) - q.log_prob(x)).mean(0)
-            actual = kl_divergence(p, q)
-            message = 'Incorrect KL({}, {}).\nExpected (monte carlo): {}\nActual (analytic): {}'.format(
-                type(p).__name__, type(q).__name__, expected, actual)
-            self.assertEqual(expected, actual, prec=0.1, message=message)
+            for sample_shape in self.sampling_range:
```

Comment:
One direct improvement I could think of is this:
1. Sample 100000 points.
2. Perform the `assertEqual` checks on slices of them.